reactor-netty-core/src/main/java/reactor/netty/BufferMono.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty;
+
+import io.netty5.buffer.BufferInputStream;
+import io.netty5.buffer.api.Buffer;
+import io.netty5.buffer.api.BufferAllocator;
+import io.netty5.buffer.api.CompositeBuffer;
+import io.netty5.buffer.api.Send;
+import org.reactivestreams.Publisher;
+import reactor.core.CoreSubscriber;
+import reactor.core.Fuseable;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoOperator;
+
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Objects;
+
+import static io.netty5.buffer.api.DefaultBufferAllocators.preferredAllocator;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A decorating {@link Mono} {@link NettyInbound} with various {@link Buffer} related
```

Comment:
is the `NettyInbound` mention still relevant? If not, the javadocs should probably be revised, as I imagine a `BufferMono` can be used for inbound or outbound operations. (it would be fine to open an issue to track that javadoc improvement separately)

Reply:
(note this also applies to `BufferFlux`)

Reply:
`NettyInbound`/`NettyOutbound` are needed.
We must work with `Publisher` interface not with `Flux`/`Mono`
also there we have method not only for receiving/sending data but working with the connection